salvo/src/lib/test_job_control_loader.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+"""
+Test module to validate parsing of the job control document
+"""
+import os
+import json
+import site
+import tempfile
+import pytest
+
+from google.protobuf.json_format import (MessageToJson)
+
+site.addsitedir("src")
+
+from job_control_loader import load_control_doc
+from api.control_pb2 import JobControl
+from api.docker_volume_pb2 import (Volume, VolumeProperties)
+
+
+def _write_object_to_disk(pb_obj, path):
+  """
+    Store a formatted json document to disk
+    """
+  json_obj = MessageToJson(pb_obj, indent=2)
+  with open(path, 'w') as json_doc:
+    json_doc.write(json_obj)
+
+  print("\n===== BEGIN ======")
+  print(json_obj)
+  print("====== END ========\n")
+
+
+def _serialize_and_read_object(pb_obj):
+  """
+    Serialize a protobuf object to disk and verify we can re-read it as JSON
+    """
+  with tempfile.NamedTemporaryFile(mode='w', delete=True) as tmp:
+    _write_object_to_disk(pb_obj, tmp.name)
+
+    with open(tmp.name, 'r') as json_file:
+      json_object = json.loads(json_file.read())
+      assert json_object is not None
+      assert json_object != {}
+
+
+def _validate_job_control_object(job_control):
+  """
+    Common verification function for a job control object
+    """
+  assert job_control is not None
+
+  # Verify execution location
+  assert job_control.remote
+
+  # Verify configured benchmark
+  assert job_control.scavenging_benchmark
+  assert not job_control.dockerized_benchmark
+  assert not job_control.binary_benchmark
+
+  # Verify sources
+  assert job_control.source is not None or job_control.source != []
+  assert len(job_control.source) == 2
+
+  saw_envoy = False
+  saw_nighthawk = False
+  for source in job_control.source:
+    if source.nighthawk:
+      assert not source.source_path
+      assert source.source_url == "https://github.com/envoyproxy/nighthawk.git"
+      assert source.branch == "master"
+      assert not source.commit_hash
+      saw_nighthawk = True
+
+    elif source.envoy:
+      assert source.source_path == "/home/ubuntu/envoy"
+      assert not source.source_url
+      assert source.branch == "master"
+      assert source.commit_hash == "e744a103756e9242342662442ddb308382e26c8b"
```

Comment:
What's the plan to deal with these hardcoded hashes? Should they be in some configuration file?

Reply:
These are just string literals with no meaning.  This could be "foo" and work the same.